### PR TITLE
Update sigrok formulas

### PR DESCRIPTION
**Pull request by denisvm**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [ ] Written a sensible test? (The best test is to compile and run a sample program.)

### Updates to existing formula: have you

- [ ] Removed the `revision` line, if any, when bumping the version number?
- [ ] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

### CODE REVIEW

File: pulseview.rb [hunk 1]
```diff
@@ -19,10 +20,11 @@ class Pulseview < Formula
 
   depends_on "pkg-config" => :build
   depends_on "cmake" => :build
+  depends_on :x11 => :build
```
Comment (iMichka) [inline pulseview.rb]: Is this for Linux only, or for mac too? If it is for Linux only, please add `unless OS.mac?` at the end of the line.

File: libsigrokdecode.rb [hunk 2]
```diff
@@ -70,6 +80,13 @@ class Libsigrokdecode < Formula
     depends_on "automake" => :build
   end
 
+  if build.with?("fx2lafw")
+    depends_on "libtool" => :build
```
Comment (iMichka) [inline libsigrokdecode.rb]: Are all these dependencies for mac, or are some Linux only? Please add unless OS.mac? at the end of the line if some of these are for Linux only.

Also; these are also added when building with head. Is that right? Could these be regrouped?

Comment (denisvm) [inline libsigrokdecode.rb]: In README it says that lib tool is required when building 'from git', but when I run `autogen.sh` it disables libtool, so it looks like it's not really required, at least on Mac. I've added your suggestion in all formulas.

Also grouped the common dependencies for HEAD in head section.

File: pulseview.rb (hunk 1 above)
Comment (denisvm) [inline pulseview.rb]: The original formula referenced it through the lines below, but it doesn't really seem to need it. So I'm removing it for now.
```c
-DPNG_INCLUDE_DIR=#{MacOS::X11.include}```

Comment (iMichka) [inline pulseview.rb]: Yes, if we can drop x11 stuff, this would be great.

**Comment by denisvm** [comment]
I tried to add a test to libsigrokdecode, but I couldn't find a way to include reference to other libraries in the system. I've commented the test block for now.

```fatal error: 'glib.h' file not found```

**Comment by iMichka** [comment]
You can use something like `Formula["glib"].opt_lib` and `Formula["glib"].opt_include` to pass the paths to the compiler. In the `test` block we do not set the paths, so it needs to be done manually.

**Comment by denisvm** [comment]
Great, thanks for the info. I've just added the library tests :)

**Comment by denisvm** [comment]
Is there anything missing to merge this or should we wait for Qt Linux bottles?

**Comment by iMichka** [comment]
@denisvm Thanks for your contribution, and for the formula cleanup!

**Pull request closed without merging**